connection.go

```diff
@@ -162,28 +165,101 @@ func (mc *mysqlConn) Prepare(query string) (driver.Stmt, error) {
 	return stmt, err
 }
 
+// https://github.com/mysql/mysql-server/blob/mysql-5.7.5/libmysql/libmysql.c#L1150-L1156
+func (mc *mysqlConn) escapeBytes(v []byte) string {
+	var escape func([]byte) []byte
+	if mc.status&statusNoBackslashEscapes == 0 {
+		escape = EscapeString
+	} else {
+		escape = EscapeQuotes
+	}
+	return "'" + string(escape(v)) + "'"
+}
+
+func (mc *mysqlConn) buildQuery(query string, args []driver.Value) (string, error) {
+	chunks := strings.Split(query, "?")
```

Comment:
To be taken with a grain of salt - this is only called once per query. Still...

I'd like to reduce the number of allocations - that's also why I sent you the playground link.
Start with a buffer: `make([]byte, 0, 5*len(query)/4)` - byte slice with 0 length and 1.25 times capacity of original query length (1.25 is just a best guess for a sensible starting point).
Use the `Append...` based functions from strconv.
Try looping with `strings.IndexByte(..., '?')` instead of splitting on '?'.